src/main/java/nom/tam/image/compression/hdu/CompressedImageHDU.java

```diff
@@ -142,6 +144,48 @@ public CompressedImageHDU(Header hdr, CompressedImageData datum) {
     }
 
     public ImageHDU asImageHDU() throws FitsException {
+        final Header header = getGenericHeader();
+        ImageData data = (ImageData) ImageHDU.manufactureData(header);
+        ImageHDU imageHDU = new ImageHDU(header, data);
+        data.setBuffer(getUncompressedData());
+        return imageHDU;
+    }
+
+     /**
+     * Given this compressed HDU, get the original (decompressed) axes.
+     *
+     * @return the dimensions of the axis.
+     * @throws FitsException
+     *             if the axis are configured wrong.
+     */
+    public int[] getDecompressedAxes() throws FitsException {
+        int nAxis = this.myHeader.getIntValue(Compression.ZNAXIS);
+        if (nAxis < 0) {
+            throw new FitsException("Negative ZNAXIS (or NAXIS) value " + nAxis);
+        }
+        if (nAxis > CompressedImageHDU.MAX_NAXIS_ALLOWED) {
+            throw new FitsException("ZNAXIS/NAXIS value " + nAxis + " too large");
+        }
+
+        if (nAxis == 0) {
+            return null;
+        }
+
+        final int[] axes = new int[nAxis];
+        for (int i = 1; i <= nAxis; i++) {
+            axes[nAxis - i] = this.myHeader.getIntValue(Compression.ZNAXISn.n(i));
+        }
+
+        return axes;
+    }
+
+    /**
+     * Obtain a header representative of a decompressed ImageHDU.
+     * @return Header with decompressed cards.
+     * @throws HeaderCardException
+     *          if the card could not be copied
+     */
+    public Header getGenericHeader() throws HeaderCardException {
```

Comment:
I wonder if this should have more limited visibility? Or do we want this exposed to the users? Also perhaps call it `getImageHeader()` or `getDescriptiveImageHeader()`?

Reply:
I exposed it so that callers too could get the decompressed Image header without decompressing the HDU.  I didn't see a better way to do that.  I'll rename it though, and I neglected to show its usage in the `README`.